storage-local/src/main/java/io/kestra/storage/local/LocalStorage.java

```diff
@@ -25,14 +23,14 @@ public class LocalStorage implements StorageInterface {
     @Inject
     public LocalStorage(LocalConfig config) {
         this.config = config;
-        this.createDirectory(null);
+        this.createParentDirectories(null);
     }
 
     private Path getPath(URI uri) {
         return Paths.get(config.getBasePath().toAbsolutePath().toString(), uri.toString());
     }
 
-    private void createDirectory(URI append) {
+    private void createParentDirectories(URI append) {
```

Comment:
Please delete this method and replace calls with createDirectory(path)
with path being null in the constructor (so make it nullsafe in which case it creates the root directory) or the parent uri directly in the put method
this will prevent having to handle such complex method and get rid of confusion around why there is 2 create directory methods